state/account_test.go

```diff
@@ -36,6 +35,15 @@ func TestNonce(t *testing.T) {
 		require.NoError(acct.SetPendingNonce(1))
 		require.Equal(uint64(1), acct.PendingNonce())
 	})
+	t.Run("legacy fresh account type", func(t *testing.T) {
+		acct, err := NewAccount(LegacyNonceAccountTypeOption())
+		require.NoError(err)
+		require.Equal(uint64(1), acct.PendingNonce())
+		require.Error(acct.SetPendingNonce(0))
+		require.Error(acct.SetPendingNonce(3))
```

Comment:
if `SetPendingNonce(2)` it will skip this conversion, and remain legacy account. 
Do we want to allow this behavior? i.e., the legacy account could convert to zero-nonce type (by sending first tx nonce=0), or remain legacy (by sending first tx nonce=1)